x-pack/elastic-agent/pkg/agent/application/upgrade/locker.go

```diff
@@ -0,0 +1,50 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package upgrade
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/gofrs/flock"
+)
+
+const watcherLockFile = "watcher.lock"
+
+// ErrAlreadyLocked is returned when lock is already taken.
+var ErrAlreadyLocked = errors.New("watcher already locked")
+
+// Locker locks the agent.lock file inside the provided directory.
+type Locker struct {
+	lock *flock.Flock
+}
+
+// NewLocker creates an Locker that locks the agent.lock file inside the provided directory.
+func NewLocker(dir string) *Locker {
```

Comment:
We have a locker already used by the application, to ensure that no more than one Agent is running. Could we generalize that code to share it? This seems very similar.

Reply:
yep it is pretty much the same i can generalize that